pkg/reconciler/domainmapping/reconciler.go

```diff
@@ -136,6 +136,11 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, dm *v1alpha1.DomainMappi
 
 // FinalizeKind cleans up the ClusterDomainClaim created by the DomainMapping.
 func (r *Reconciler) FinalizeKind(ctx context.Context, dm *v1alpha1.DomainMapping) reconciler.Event {
+	if !config.FromContext(ctx).Network.AutocreateClusterDomainClaims {
+		// If we're not responsible for creating domain claims, we're not responsible for cleaning them up.
```

Comment:
what if I created DM and then disabled the flag? Should we still cleanup the existing ones?

Reply:
I think no, if the flag is disabled that implies the administrator is going to manage creating/deleting claims.